### PR TITLE
Enable engine postgres debug builds

### PR DESCRIPTION
**Pull request by gruuya**
Add Dockerfiles to: 

1. Compile/build/install PG with debugging capabilities and libs, 
2. Build the engine on top of that.

The debug flow with the new makefile commands is:
- Build PG debug base image: `make build.pg_debug`
- Build engine image out of it: `make build.debug`
- Create engine container: `sgr engine add -i splitgraph/engine_debug:development --password test --no-pull --cap-add SYS_PTRACE`
- Open engine DB shell ` pgcli $(sgr config -n)`, and get client pid `SELECT pg_backend_pid()`
- Enter engine container `docker exec -ti splitgraph_engine_default bash`
- Attach gdb to process: `gdb -p <pid>` (The first continue command will fail, but the next one will succeed).
- [Profit](https://www.highgo.ca/2019/10/03/trace-query-processing-internals-with-debugger/)

CU-1t1wycg

### CODE REVIEW

File: engine/Dockerfile.pg_debug [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# prefix (PGXS's feature) and then copying that over into the filesystem root in the target.
+
+
+#####
+##### toolchain
+#####
+
+FROM postgres:12.3
+
+RUN apt-get update -qq && \
```
Comment (mildbyte) [inline engine/Dockerfile.pg_debug:17]: I feel like most of these apt packages (e.g. libmongoc) aren't needed, can you unify these two apt calls?

File: engine/Dockerfile.pg_debug [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# syntax = docker/dockerfile:experimental
+
+# Multistage Dockerfile to build the Splitgraph engine: because of all the compilation of
```
Comment (mildbyte) [inline engine/Dockerfile.pg_debug:3]: Redundant comment

**Comment by mildbyte** [comment]
Left some comments; the pg_debug Dockerfile could use some cleanup to remove copypaste leftovers from the standard engine image but LGTM otherwise!

File: engine/Dockerfile.pg_debug (hunk 1 above)
Comment (gruuya) [reply engine/Dockerfile.pg_debug:17]: Sure thing, let me see what's redundant.

**Pull request merged**